### ISSUE TITLE
Another models

### ISSUE THREAD

**Issue by wzhhasadream** [freedbee/Neuron_Descent_and_KFAC#1]
Hello, I'm reproducing your experiment. After my test, FOOF performs well in the training set, but not very well in the test set. In addition, do your models support the Resnet structure? Thank you

**Comment by freedbee** [comment]
Hi, 

thanks a lot for your interest in the paper and code!

(1) Re: Test Set results
- In my experiments, on the test set, FOOF performed slightly better than SGD on Fashion MNIST and slightly worse than SGD on MNIST -- so results weren't conclusive. Much more important is the following point:
- For test/validation results to be meaningful, you need to do at least two things. (a) Tune HPs to maximize validation performance. **(b) Add and tune regularisation.** An instructive example of how important this is can be found in Figure 1 [here](https://arxiv.org/abs/1810.12281): Without regularisation, KFAC generalises worse than SGD, but with regularisation it generalises better (regularisation being as simple as weight decay in this case). So, as stated in the discussion of the paper, with the current experimental setup it is not possible to make meaningful statements about generalisation performance.

(2) Re: ResNet
I added code for the resnet. I briefly checked it and it seems to work as intended. I don't currently have time to be super thorough, so let me know if anything unexpected comes up.

**Comment by wzhhasadream** [comment]
Thank you very much for your help
